apps/mobile/src/features/notifications/use-notifications.ts

```diff
@@ -0,0 +1,167 @@
+import { useCallback, useEffect } from 'react';
+import { PermissionsAndroid, Platform } from 'react-native';
+
+import { useToastContext } from '@/components/toast/toast-context';
+import { useReleasePushNotificationsFlag } from '@/features/feature-flags/use-feature-flags';
+import { useRegisterNotificationMutation } from '@/queries/notifications/notifications.query';
+import { useBitcoinAddresses } from '@/store/keychains/bitcoin/bitcoin-keychains.read';
+import { useStacksSignerAddresses } from '@/store/keychains/stacks/stacks-keychains.read';
+import { useSettings } from '@/store/settings/settings';
+import messaging from '@react-native-firebase/messaging';
+import * as Notifications from 'expo-notifications';
+
+interface RegisterNotifications {
+  /**
+   * TODO: Unregister FCM token. New notifications sent to this token would fail.
+   * We should use this endpoint when user resets the app (deletes all wallets)
+   *
+   */
+  unregisterNotifications(): Promise<void>;
+
+  /** Registers notifications for BTC and STX addresses in our backend.
+   *
+   * /register endpoint is going to work as a sync for addresses connected to this device (across multiple wallets).
+   * We need to call it when user:
+   * 1.Adds an account
+   * 2.Adds a wallet
+   * 3.Deletes a wallet
+   * 4. Toggles the notification setting
+   *
+   */
+  registerNotifications(): Promise<void>;
+
+  requestUserPermissions(): Promise<void>;
+
+  registerNotificationsBTC(): Promise<void>;
+  registerNotificationsSTX(): Promise<void>;
```

Comment:
```suggestion
  registerNotificationsBtc(): Promise<void>;
  registerNotificationsStx(): Promise<void>;
```


https://www.notion.so/trustmachines/Naming-Camelcase-acronyms-119d74b694c78080974aee5e3bc18bc9?pvs=4